### PR TITLE
chore: remove deprecated apis and ENABLE_BROAD_ACTIVITY_ACCESS

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Removes old deprecated APIs:
- `/superset/recent_activity/...` 
- `/superset/fave_dashboards_by_username/...`
- `/superset/fave_dashboards/...`
- `/superset/created_dashboards/...`
- `/superset/user_slices/...`
- `/superset/created_slices/...`
- `/superset/fave_slices/...`
- `/superset/favstar/...`

This PR also removes `ENABLE_BROAD_ACTIVITY_ACCESS` feature flag (was disabled by default), since by removing the above mentioned deprecated API's only one final API ( `/api/v1/log/recent_activity/` ) was using it.
Also removes `changed_by_url` and `created_by_url` REST API response fields from datasets, dashboards and charts, since these fields function was to provide a link to other user's profiles.

Effort on removing deprecated APIs from `/superset`: https://github.com/apache/superset/discussions/24332

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1363,27 +1087,17 @@ def welcome(self) -> FlaskResponse:
...
         payload = {
-            "user": bootstrap_user_data(user, include_perms=True),
+            "user": bootstrap_user_data(g.user, include_perms=True),
             "common": common_bootstrap_payload(g.user),
         }
 
         return self.render_template(
             "superset/basic.html",
-            title=_("%(user)s's profile", user=username).__str__(),
+            title=_("%(user)s's profile", user=g.user.username).__str__(),
```
Comment (john-bodley) [inline superset/views/core.py:1100]: Maybe 

```suggestion
            title=_("%(user)s's profile", user=get_username()).__str__(),
```

for consistency as we use that elsewhere. That said I guess you're already using `g.user` previously.

File: superset/views/core.py [hunk 2]
```diff
@@ -1363,27 +1087,17 @@ def welcome(self) -> FlaskResponse:
...
-        # Prevent returning 404 when user is not found to prevent username scanning
-        user_id = -1 if not user else user.id
-        # Prevent unauthorized access to other user's profiles,
-        # unless configured to do so with ENABLE_BROAD_ACTIVITY_ACCESS
-        if error_obj := self.get_user_activity_access_error(user_id):
-            return error_obj
-
         payload = {
-            "user": bootstrap_user_data(user, include_perms=True),
+            "user": bootstrap_user_data(g.user, include_perms=True),
```
Comment (john-bodley) [inline superset/views/core.py:1094]: @dpgaspar what happens if the user is a guest user? Is `g.user` guaranteed to exist and be non-`None`.

File: superset/views/log/api.py [hunk 3]
```diff
@@ -125,8 +125,7 @@ def recent_activity(self, user_id: int, **kwargs: Any) -> FlaskResponse:
             500:
               $ref: '#/components/responses/500'
         """
-        if error_obj := self.get_user_activity_access_error(user_id):
-            return error_obj
+        user_id = get_user_id()
```
Comment (john-bodley) [inline superset/views/log/api.py:128]: Maybe inline this on line 136? Alternatively given it seems like this is the only place which calls it should we augment the [get_recent_activity](https://github.com/apache/superset/blob/d2b0b8eac52ad8b68639c6581a1ed174a593f564/superset/views/log/dao.py#L36-L38) to not include the user ID, i.e., the recently activity always pertains to the current user? That's also likely safer (less vulnerable) from a security perspective.

**Review by john-bodley — COMMENTED** [review]
Thanks for the cleanup. I left a couple of comments, but otherwise LGTM.

File: superset/views/core.py (hunk 2 above)
Comment (dpgaspar) [reply superset/views/core.py]: Good point, it should be `AnonymousUser` from flask-login or `GuestUser`, but added some safe guard, and we're returning 404 now

**Review by villebro — APPROVED** [review]
LGTM

File: superset/views/core.py [hunk 4]
```diff
@@ -1363,27 +1076,20 @@ def welcome(self) -> FlaskResponse:
...
-            db.session.query(ab_models.User).filter_by(username=username).one_or_none()
-        )
-        # Prevent returning 404 when user is not found to prevent username scanning
-        user_id = -1 if not user else user.id
-        # Prevent unauthorized access to other user's profiles,
-        # unless configured to do so with ENABLE_BROAD_ACTIVITY_ACCESS
-        if error_obj := self.get_user_activity_access_error(user_id):
-            return error_obj
-
+        user = g.user if hasattr(g, "user") and g.user else None
```
[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/views/core.py:1082]: totally agree! would be great also to make anonymous users and guest user's more clear to check, use and test

[the remark this answers is not in the document]
Comment (john-bodley) [reply superset/views/core.py:1082]: @villebro having a `get_user` method [here](https://github.com/apache/superset/blob/892de8a232a5b40dd22bf07786cb7255ef633d44/superset/utils/core.py#L1398-L1426)—which `get_user_id` and `get_username` would call—would be useful.

**Review by john-bodley — APPROVED** [review]
LGTM

**Pull request merged**